src/abc_inference.jl

```diff
@@ -1,22 +1,22 @@
-function createabcfunction(prob, t, distancefunction, alg, kwargs...)
-    function simfunc(params, constants, targetdata)
-        sol = solve(problem_new_parameters(prob, params), alg, saveat = t, save_start = false, kwargs...)
-        data = convert(Array, sol)
-        distancefunction(targetdata, data), data
+function createabcfunction(prob, t, distancefunction, alg; kwargs...)
+    function simfunc(params, constants, data)
+        sol = solve(problem_new_parameters(prob, params), alg; saveat = t, save_start = false, kwargs...)
```

Comment:
Using this with callbacks showed that setting the `saveat` option is very brittle as often other time-points are also saved (such as the time of the event).  I'll update it to just use interpolation.  Probably applies to other solvers too.

Reply:
Yeah, that makes a lot of sense. These examples might work then if the user does `save_positions = (false,false)`. I think that an option to override callback saving may be worthwhile? I am not sure though. Another issue I've had with the current strategy is that it's not mutable. I think DiffEq.jl may need an issue on improving callback saving strategies.

Reply:
But for this I would, is should not matter too much to use interpolation, unless the solve is very cheap, right?

Reply:
Yeah. The post-solution interpolation would just require some extra searches and have an extra allocation, but if the solve is expensive enough this wouldn't even be noticed and would be more robust. I can't say I've checked where the cutoff is, for example, what overhead would that add to Lotka-Volterra? No idea.

Reply:
Ok if I push a fix for this too to this PR?  I can do that now, and then we can merge.

Reply:
An I will squash into one commit.

Reply:
@mauro3 `abc_inference` doesn't face problems in case of callbacks, right?

Reply:
I would like to get a sense of the cost from doing it first.

Reply:
Ok, let's merge it like this then and investigate the `saveat` in another PR.  In the problem I'm currently working with, which has callbacks, I had to modify it. This is what I use:
```julia
function createabcfunction(prob, t, distancefunction, alg; kwargs...)
    function simfunc(params, constants, data)
        sol = solve(problem_new_parameters(prob, params), alg; kwargs...)
        if any((s.retcode != :Success for s in sol))
            return Inf, nothing
        else
            simdata = convert(Array, sol(t))
            distancefunction(data, simdata), nothing
        end
    end
end
```

Maybe I ran the example in #43 with my fix, as Chris had ABC also fail there.